File: PubLibIS.DAL/Models/Periodical.cs
using Dapper.Contrib.Extensions;
using PubLibIS.DAL.Enums;
using System;
using System.Collections.Generic;

namespace PubLibIS.DAL.Models
{
    [Table("Periodicals")]
    public class Periodical : BaseEntity
    {
        public Periodical()
        {
            PeriodicalEditions = new List<PeriodicalEdition>();
        }

        public string ISSN { get; set; }
        public PeriodicalType Type { get; set; }
        public string Name { get; set; }
        public DateTime Foundation { get; set; }
        public bool IsPublished { get; set; }
        public int? PublishingHouse_Id { get; set; }

        [Write(false)]
        public virtual PublishingHouse PublishingHouse { get; set; }
        [Write(false)]
        public virtual ICollection<PeriodicalEdition> PeriodicalEditions { get; set; }
    }
}

File: PubLibIS.CoreUI/src/app/models/periodical.ts
import { PublishingHouse } from "./publishingHouse";
import { PeriodicalType } from "./periodicalType";

export class Periodical {
  constructor(
    public id?: number,
    public issn?: string,
    public name?: string,
    public type?: PeriodicalType,
    public foundation?: string,
    public publishingHouse_Id?: number[],
    public publicationsCount?: number,
    public publishingHouse?: PublishingHouse) { }
}
